imports/wasi_snapshot_preview1/wasi_stdlib_test.go

```diff
@@ -336,7 +341,44 @@ func testOpen(t *testing.T, cmd string, bin []byte) {
 			WithArgs("wasi", "open-"+cmd).
 			WithFSConfig(wazero.NewFSConfig().WithDirMount(t.TempDir(), "/"))
 
-		console := compileAndRun(t, moduleConfig, bin)
+		console := compileAndRun(t, testCtx, moduleConfig, bin)
 		require.Equal(t, "OK", strings.TrimSpace(console))
 	})
 }
+
+func Test_Sock(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("windows is not supported yet")
+	}
+	// Create a listener to get a random available port.
+	ln, err := net.Listen("tcp", "0.0.0.0:0")
+	require.NoError(t, err)
+	// Ensure that the address is correctly handled in CI.
+	// GitHub Actions may not hand over an IPv6 Address,
+	// so we force the IPv4 loopback explicitly.
+	port := ln.Addr().(*net.TCPAddr).Port
+	addr := fmt.Sprintf("127.0.0.1:%d", port)
+	// Close the listener and let it wazero create it instead.
+	ln.Close()
+	time.Sleep(1 * time.Second)
+	moduleConfig := wazero.NewModuleConfig().WithArgs("wasi", "socket").WithSysNanosleep()
```

Comment:
we should have a comment why this needs real sleep to work

Reply:
I suspect it was another attempt to "fix" CI. I can remove it locally. We can try to push again without it and see what happens